nft_ingester/src/bin/slot_persister/main.rs

```diff
@@ -497,3 +559,261 @@ async fn process_slots(
+                    let mut missing_slots_lock = missing_slots.lock().unwrap();
+                    missing_slots_lock.push(slot);
+                    missing_slots_lock.extend(slots_iter);
+                }
+                break;
+            }
+        }
+
+        // Print missing slots
+        let missing_slots = Arc::into_inner(missing_slots).unwrap().into_inner().unwrap();
+        Some((lower_slot_boundary, last_valid_key, missing_slots))
+    }
+
+    let in_mem_dumper = Arc::new(InMemorySlotsDumper::new());
+    let slots_collector =
+        SlotsCollector::new(in_mem_dumper.clone(), backfill_source.clone(), metrics.clone());
+    let slot_backfill_offset = args.slot_backfill_offset;
+    loop {
+        let db = target_db.clone();
+        let dumper = in_mem_dumper.clone();
+        let collector = slots_collector.clone();
+        let (last_checkpoint, next_checkpoint, missed_slots) = get_missed_slots(
+            dumper,
+            collector,
+            db,
+            slot_backfill_offset,
+            cancellation_token.child_token(),
+        )
+        .await
+        .expect("retrieval of missed slots must join");
+        if !missed_slots.is_empty() {
+            debug!(?missed_slots, "Missed slots found");
+            info!(%last_checkpoint, %next_checkpoint, "Getting missed slots between checkpoints");
+            process_slots(
+                missed_slots.clone(),
+                backfill_source.clone(),
+                target_db.clone(),
+                &args,
+                cancellation_token.child_token(),
+            )
+            .await;
+            let missed_slots_map = missed_slots
+                .into_iter()
+                .map(|slot| {
+                    let next_seq = target_db
+                        .next_missed_slots_seq()
+                        .expect("get next missed slots seq must not fail");
+                    ((next_seq, slot), MissedSlotsIdx)
+                })
+                .collect();
+            debug!("Saving missed slots to the database");
+            target_db
+                .missed_slots_idx
+                .put_batch(missed_slots_map)
+                .await
+                .expect("must put batch of missed slots into ");
+        }
+        debug!(%last_checkpoint, "Deleting last consistency checkpoint");
+        target_db
+            .slot_consistency_checkpoint
+            .delete(last_checkpoint)
+            .expect("must delete the previous checkpoint after slot processing");
+        debug!(%next_checkpoint, "Setting the new checkpoint value");
+        target_db
+            .slot_consistency_checkpoint
+            .put(next_checkpoint, SlotConsistencyCheckpoint)
+            .expect("must update the new checkpoint after the slots are saved into storage");
+        tokio::select! {
+            _ = cancellation_token.cancelled() => { break; }
+            _ = tokio::time::sleep(Duration::from_secs(args.slot_backfill_interval)) => {
```

Comment:
Are we ok with the initial check taking 50 days? 200M slots of range with 1M every 6 hours will take long

Reply:
this was tested with 1M range every 120 seconds and worked fine; i will lower the limits drastically.